src/utils/index.test.ts

```diff
@@ -25,54 +26,151 @@ describe('Utils', () => {
       );
     });
 
     it('should construct a HTML element overwriting default attribute values with user-defined inputs', () => {
       const element = 'lite-element';
-      const attrs = {
+      const defaultAttrs = {
         id: '123',
         src: {
           url: 'https://example.com/',
           params: ['unit', 'type'],
         },
       };
-      const args = {
+      const htmlAttrInputs = {
         src: 'https://example.com/overwrite',
       };
 
-      const htmlElement = createHtml(element, attrs, args);
+      const htmlElement = createHtml(element, defaultAttrs, htmlAttrInputs);
       expect(htmlElement).toEqual(
         '<lite-element id="123" src="https://example.com/overwrite"></lite-element>',
       );
     });
   });
+
+  describe('formatData', () => {
+    it('should correctly format and overwrite data and inputs', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            src: {
+              url: 'https://www.example.com/',
+              params: ['id'],
+            },
+            width: '100',
+            height: '100',
+          },
+        },
+      };
+
+      const inputs = {
+        id: '123',
+        loading: 'auto',
+        width: '150',
+      };
+
+      const result = formatData(data, inputs);
+      expect(result.html).toEqual(
+        '<iframe loading="auto" src="https://www.example.com/?id=123" width="150" height="100"></iframe>',
+      );
+      expect(result.scripts).toEqual(null);
+    });
+
+    it('should pass scripts and correctly assign params if available', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+          },
+        },
+        scripts: [
+          {
+            url: 'https://www.example.com',
+            params: ['id'],
+            strategy: 'worker',
+            location: 'head',
+            action: 'append',
+          },
+        ],
+      };
+
+      const inputs = {
+        id: '123',
+      };
+
+      const result = formatData(data as Data, inputs);
+      expect(result.html).toEqual('<iframe loading="lazy"></iframe>');
+      expect(result.scripts).not.toEqual(null);
+      expect(result.scripts!.length).toEqual(1);
+      expect(result.scripts![0].url).toEqual('https://www.example.com/?id=123');
+    });
+
+    it('should forward all additional inputs as html attributes if not used elsewhere', () => {
+      const data = {
+        id: 'third-party',
+        description: 'Description',
+        html: {
+          element: 'iframe',
+          attributes: {
+            loading: 'lazy',
+            width: '100',
+            height: '100',
+          },
+        },
+      };
+
+      const inputs = {
+        id: '123',
```

Comment:
If we pass in inputs like `{ id: "props.id" }` will that be inlined the same? We should highlight that in the test to make it clear